azurerm/internal/services/managedapplication/validate/managed_application_definition_test.go

```diff
@@ -0,0 +1,147 @@
+package validate
+
+import (
+	"testing"
+)
+
+func TestManagedApplicationDefinitionName(t *testing.T) {
+	testData := []struct {
+		input    string
+		expected bool
+	}{
+		{
+			input:    "",
+			expected: false,
+		},
+		{
+			input:    "h",
+			expected: false,
+		},
+		{
+			input:    "he",
+			expected: false,
+		},
+		{
+			input:    "hel",
+			expected: true,
+		},
+		{
+			input:    "hel2",
+			expected: true,
+		},
+		{
+			input:    "_hello",
+			expected: false,
+		},
+		{
+			input:    "hello-",
+			expected: false,
+		},
+		{
+			input:    "malcolm-in!the-middle",
+			expected: false,
+		},
+		{
+			input:    "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijk",
```

Comment:
Can we use `strings.Repeat()` here, so as to be easier tell how many characters there are (same for other places in this file).

Reply:
updated